tests/samples/OpaqueTypes.h

```diff
@@ -0,0 +1,22 @@
+struct point;
+struct point *move(struct point *point, int x, int y);
+
+typedef struct points points;
+typedef union u u;
```

Comment:
How about declaring u before points before point and have u use points which uses point? It would test that we generate types in the right order.

Reply:
It will generate types in the order they appear in a header file.
It might be not a trivial task to sort them.

So in this example:
```c
typedef struct points points;
typedef struct point point;

struct points {
    point *point1;
    point *point2;
};

struct point {
    int x;
    int y;
};
```
`points` appear before `point`. To sort them we need to find out that `points` actually uses `point`.
As I see it now we need to check for each type if it uses each other type and then do topological sort.
Tricky part is that there might be circular dependencies:
```c
struct b;
struct c;

struct a {
   struct b *b;
};

struct b {
    struct c *c;
};

struct c {
    struct a *a;
};
```

I will create an issue about it